Initial Pull request for unit tests and other enhancements required for swagger documenation

rajeshwari16945 opened the pull request:
Update project structure and add unit tests
- Enhanced `.gitignore` to exclude unnecessary files.
- Added `Microsoft.EntityFrameworkCore` package reference.
- Modified `TimeEntriesController` to always return true for holiday checks.
- Updated project paths in `ClockIn.csproj.nuget.dgspec.json`.
- Regenerated `ClockIn.AssemblyInfo.cs` and updated build properties.
- Created `ClockIn.UnitTest` project with necessary testing packages.
- Added `appsettings.json` for logging configuration.
- Implemented new controllers and unit tests for CRUD operations.
- Updated `Program.cs` to configure Swagger for API documentation.

rajeshwari16945 committed:
Update project structure and add unit tests

- Enhanced `.gitignore` to exclude unnecessary files.
- Added `Microsoft.EntityFrameworkCore` package reference.
- Modified `TimeEntriesController` to always return true for holiday checks.
- Updated project paths in `ClockIn.csproj.nuget.dgspec.json`.
- Regenerated `ClockIn.AssemblyInfo.cs` and updated build properties.
- Created `ClockIn.UnitTest` project with necessary testing packages.
- Added `appsettings.json` for logging configuration.
- Implemented new controllers and unit tests for CRUD operations.
- Updated `Program.cs` to configure Swagger for API documentation.

rajeshwari16945 committed:
Approval Controller Test name change

rajeshwari16945 committed:
Add unit tests for various controllers and update project file

- Updated `ClockIn.UnitTest.csproj` to exclude `AntiForgeryControllerTests.cs` from compilation and include it as a non-compiled file.
- Updated `AssemblyInformationalVersion` in `ClockIn.AssemblyInfo.cs`.
- Added unit tests for the following controllers:
  - `AntiForgeryController`
  - `AuditLogsController`
  - `AuthController`
  - `BillingEntriesController`
  - `HolidaysController`
  - `LeavesController`
  - `ProductivityController`
  - `ProjectsController`
  - `TagsController`
  - `TaskItemsController`
  - `TeamsController`
  - `TimeEntryTagsController`
  - `UsersController`

rajeshwari16945 committed:
Remove IsHolidayOrWeekend method from controller

The `Delete` method in the `TimeEntriesController` class has been updated to eliminate the `IsHolidayOrWeekend` method. This method, which checked if a date was a holiday or weekend, was previously commented out and is now fully removed. The functionality of the `Delete` method remains unchanged.

rajeshwari16945 committed:
Update .gitignore and remove assembly info files

- Added `project.fragment.lock.json` and generated files
  to `.gitignore`.
- Removed all content from `ClockIn.AssemblyInfo.cs`,
  including assembly metadata.
- Deleted `ClockIn.AssemblyInfoInputs.cache` file.

rajeshwari16945 committed:
Refactor authorization and user management features

- Commented out antiforgery token handling in AntiForgeryController.
- Added role-based authorization to multiple controllers.
- Updated AuthController for user validation and registration.
- Refactored User model to include FirstName, LastName, and PasswordHash.
- Moved repository interfaces to a separate namespace.
- Adjusted SQL insert statements to rely on database-generated IDs.
- Updated LoginRequest to use Email instead of Username.
- Enhanced time_entries table schema with new columns for status, billable, and total_hours.

rajeshwari16945 committed:
Add repository interfaces and refactor user model tests

- Added `using ClockIn.DataLayer.IRepositories` to multiple test files to utilize new repository interfaces.
- Updated `AddAsync` setup in `TimeEntryTagsControllerTests.cs` to ensure proper task handling.
- Refactored `User` model in `UsersControllerTests.cs` by replacing `Name` with `FirstName`, affecting multiple test methods.
- Overall improvements enhance test structure and align with updated data model.

rajeshwari16945 committed:
Update README.md for improved clarity and details

Enhanced the description of the TimeTrackingServices application, including features, technologies, and project structure. Added sections for prerequisites, installation steps, running tests, API documentation, contributing guidelines, and contact information. Improved formatting for readability and corrected the connection string example.

SharanChaitanyaReddy commented on ClockIn/ClockIn.csproj:
```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
   <PackageReference Include="Dapper" Version="2.1.66" />
   <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="8.0.18" />
+  <PackageReference Include="Microsoft.EntityFrameworkCore" Version="9.0.7" />
```
@rajeshwari16945  we are using dapper  in this project not entity Framework. Why do we need entity Framework ?

SharanChaitanyaReddy commented on ClockIn/obj/ClockIn.csproj.nuget.dgspec.json:
This file needs to be ignored from pushing into repo, this is specific to our local.

SharanChaitanyaReddy commented on ClockIn/obj/ClockIn.csproj.nuget.g.props:
This file needs to be ignored from pushing into repo, this is specific to our local.

rajeshwari16945 committed:
Update project configuration and refactor AuthController

- Enhanced `.gitignore` to include Visual Studio and build files.
- Updated `AuthControllerTests.cs` with new dependencies and async methods.
- Created/updated `appsettings.Development.json` for logging settings.
- Removed `Microsoft.EntityFrameworkCore` from `ClockIn.csproj`.
- Refactored `AuthController` for async login and improved JWT generation.
- Deleted unnecessary `ClockIn.assets.cache` file.

SharanChaitanyaReddy commented on ClockIn/obj/ClockIn.csproj.nuget.dgspec.json:
```diff
@@ -1,93 +1,103 @@
+            },
+            "Microsoft.EntityFrameworkCore": {
+              "target": "Package",
+              "version": "[9.0.7, )"
```
remove entity framework reference

SharanChaitanyaReddy reviewed (approved):
Reviewed Unit Tests
Reviewed  null references

Looks good !
